### PR TITLE
Create Login Status using a cookie session

### PR DESCRIPTION
**Pull request by fadyawan**

### CODE REVIEW

File: react-app/src/config.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
```
Comment (Ripraskel) [inline react-app/src/config.js:1]: Not required?

File: react-app/src/Pages/RegisterPage/RegisterPage.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ function RegisterPage() {
     const registerDetails = async (e) => {
         try {
             console.log('register')
-            await axios.post('http://localhost:3001/auth/register', { username, password });
+            await axios.post('http://localhost:3001/auth/register', { username, password} );
```
Comment (Ripraskel) [inline react-app/src/Pages/RegisterPage/RegisterPage.js:14]: Change not required, and would fail most linting.

File: express-app/src/index.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
+const express = require('express');
```
Comment (Ripraskel) [inline express-app/src/index.js:1]: As discussed this needs to be broken up into multiple folders/files/functions.

File: express-app/src/index.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+const mongoose = require('mongoose');
+const bodyParser = require('body-parser');
+const cors = require('cors');
+const crypto = require('crypto');
+const { validationResult } = require('express-validator');
+const cookie = require('cookie');
+require('dotenv').config();
```
Comment (Ripraskel) [inline express-app/src/index.js:8]: Is this doing anything?

File: express-app/src/index.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    required: true,
+  },
+  salt: {
+    type: String,
+    required: true,
+  },
+}));
+
+function hashPassword(password, salt) {
+  const key = crypto.pbkdf2Sync(password, salt + config.secretKey, 1000, 64, 'sha512');
```
Comment (Ripraskel) [inline express-app/src/index.js:40]: Can you explain the value of adding a secret to the salt?

**Review by Ripraskel — CHANGES_REQUESTED** [review]
Main thing to address is to break up index.js

File: express-app/src/config.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+    secretKey: process.env.SECRET_KEY || 'defaultSecretKey',
```
Comment (Ripraskel) [inline express-app/src/config.js:2]: Should this be BIG_SECRET to align with Docker compose name?

File: react-app/src/config.js (hunk 1 above)
Comment (fadyawan) [reply react-app/src/config.js:1]: Yes

File: react-app/src/Pages/RegisterPage/RegisterPage.js (hunk 2 above)
Comment (fadyawan) [reply react-app/src/Pages/RegisterPage/RegisterPage.js:14]: Fixed

File: express-app/src/index.js (hunk 3 above)
Comment (fadyawan) [reply express-app/src/index.js:1]: Done

File: express-app/src/index.js (hunk 4 above)
Comment (fadyawan) [reply express-app/src/index.js:8]: Not after changes

File: express-app/src/index.js (hunk 5 above)
Comment (fadyawan) [reply express-app/src/index.js:40]: It only allows the access to the salt when you know the secret which improves the security.

File: express-app/src/config.js (hunk 6 above)
Comment (fadyawan) [reply express-app/src/config.js:2]: Umm not sure, will have to check.

**Pull request merged**